src/Raven.Studio/typescript/common/queryCompleter.ts

```diff
@@ -0,0 +1,279 @@
+                    break;
+                case "identifier":
+                    identifier = token.value;
+                    break;
+                case "string":
+                    const indexName = token.value.substr(1, token.value.length - 2);
+                    identifier = indexName;
+                    break;
+                case "paren.lparen":
+                    paren++;
+                    break;
+                case "paren.rparen":
+                    paren--;
+                    break;
+                case "text":
+                    if (!identifier) {
+                        text = token.value;
+                    }
+                    break;
+            }
+        } while (iterator.stepBackward());
+
+        return [null, null, null, null, null];
+    }
+
+    complete(editor: AceAjax.Editor,
+             session: AceAjax.IEditSession,
+             pos: AceAjax.Position,
+             prefix: string,
+             callback: (errors: any[], worldlist: autoCompleteWordList[]) => void) {
+
+        const [lastKeyword, operator, identifier, text, paren] = this.getLastKeyword(session, pos);
+        if (!lastKeyword)
+            return;
+
+        switch (lastKeyword) {
+            case "from": {
+                if (identifier && text) {
+                    if (paren > 0) {
+                        // from (Collection, {show fields here})
+                        this.getIndexFields(session)
+                            .done((indexFields) => callback(null, indexFields.map(field => {
+                                return {name: field, value: field, score: this.defaultScore, meta: "field"};
+                            })));
+                        return;
+                    }
+
+                    return;
+                }
+
+                if (!prefix ||
+                    prefix.length === 0 ||
+                    prefix.startsWith("@")) {
+                    callback(null, [{name: "@all_docs", value: "@all_docs", score: this.defaultScore * 10, meta: "collection"}]);
+                    callback(null, [{name: "@system", value: "@system", score: this.defaultScore - 1, meta: "collection"}]);
+                }
+                callback(null, this.collectionsTracker.getCollectionNames().map(collection => {
+                    return {
+                        name: collection,
+                        value: collection,
+                        score: this.defaultScore,
+                        meta: "collection"
+                    };
+                }));
+                break;
+            }
+            case "index": {
+                if (identifier && text) { // index name already specified
+                    return;
+                }
+
+                callback(null,
+                    this.indexes().map(index => {
+                        const name = `'${index.Name}'`;
+                        return {name: name, value: name, score: this.defaultScore, meta: "index"};
```

Comment:
The value that you see in the auto complete is the value that will be added. You cannot modify the value before it is added.